contracts/contracts/GrantRoundManager.sol

```diff
@@ -4,23 +4,23 @@ pragma solidity ^0.8.5;
 import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
 import "@openzeppelin/contracts/utils/Address.sol";
 import "@uniswap/v3-periphery/contracts/interfaces/ISwapRouter.sol";
+import "./libraries/BytesLib.sol";
 import "./GrantRegistry.sol";
 import "./GrantRound.sol";
 
 contract GrantRoundManager {
   using Address for address;
+  using BytesLib for bytes;
   using SafeERC20 for IERC20;
 
-  /// @notice Donation inputs and Uniswap V3 swap inputs: https://docs.uniswap.org/protocol/guides/swaps/single-swaps
+  /// @notice Donation inputs and Uniswap V3 swap inputs: https://docs.uniswap.org/protocol/guides/swaps/multihop-swaps
   struct Donation {
     uint96 grantId; // grant ID to which donation is being made
     GrantRound[] rounds; // rounds against which the donation should be counted
-    IERC20 tokenIn; // token in which the user made the donation
-    uint24 fee; // selected fee tier
+    bytes path; // swap path, or if user is providing donationToken, the address of the donationToken
```

Comment:
would the address of the donationToken be set as bytes ?

Reply:
Yep, but ethers will handle the conversion for us behind the scenes, so all you need to do is specify the `path` as the token address as shown in L194 here https://github.com/dcgtc/dgrants/blob/45dfae3d110c8935b9549dac94e8883d75c17c91/contracts/test/GrantRoundManager.test.ts#L186-L196